Chessington.GameEngine/ExploreInOneDirection.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Chessington.GameEngine.Tests.Pieces
+{
+    public class ExploreInOneDirection
+    {
+        public static IEnumerable<Square> ExploreOneDirection(Board board, Square initialSquare, int direction)
+        {
+            var squareList = new List<Square>();
+            var nextSquare = iterator(initialSquare, direction);
+            
+            while (true) {
```

Comment:
you can have `!CheckPosition(nextSquare.Row, nextSquare.Col, board` in your while statement